c10/core/ScalarType.h

```diff
@@ -87,7 +90,8 @@ namespace c10 {
   _(int64_t, Long, i)                         \
   _(at::Half, Half, d)                        \
   _(float, Float, d)                          \
-  _(double, Double, d)
+  _(double, Double, d)                        \
+  _(at::BFloat16, BFloat16, d)
```

Comment:
same here.

Reply:
could kill this one also?

Reply:
actually i was wrong, it is used. Filed an [issue](https://github.com/pytorch/pytorch/issues/22668) for bool and assigned myself.

Reply:
Closing the issue :) In Scalar.h there is overloaded boolean version. The overload needed because of issue with the SymbolicVariable. There is a comment: 

> "Value* is both implicitly convertible to SymbolicVariable and bool which causes ambiguosity error. Specialized constructor for bool resolves this problem."